### PR TITLE
Batch2 workchunk states hapi ready to reduce

### PR DESCRIPTION
**Pull request by TipzCM**

### CODE REVIEW

File: hapi-fhir-storage-batch2/src/test/java/ca/uhn/fhir/batch2/coordinator/ReductionStepExecutorServiceImplTest.java [hunk 1]
```diff
@@ -74,13 +70,16 @@ public void before() {
...
 
+	@SuppressWarnings({"unchecked", "rawtypes"})
 	@Test
 	public void doExecution_reductionWithChunkFailed_marksAllFutureChunksAsFailedButPreviousAsSuccess() {
 		// setup
 		List<String> chunkIds = Arrays.asList("chunk1", "chunk2");
 		List<WorkChunk> chunks = new ArrayList<>();
 		for (String id : chunkIds) {
-			chunks.add(createWorkChunk(id));
+			WorkChunk chunk = createWorkChunk(id);
```
Comment (michaelabuckley) [inline hapi-fhir-storage-batch2/src/test/java/ca/uhn/fhir/batch2/coordinator/ReductionStepExecutorServiceImplTest.java:80]: suggestion: can we have a test that uses the old states too?  Or mixes them to ensure backwards compatibility?

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImpl.java [hunk 2]
```diff
@@ -644,7 +645,7 @@ public boolean advanceJobStepAndUpdateChunkStatus(String theJobInstanceId, Strin
 					theJobInstanceId,
 					theNextStepId,
 					List.of(WorkChunkStatusEnum.GATE_WAITING, WorkChunkStatusEnum.QUEUED),
-					WorkChunkStatusEnum.READY);
+				nextStep);
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImpl.java:648]: This looks clean.

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImplTest.java [hunk 3]
```diff
@@ -489,7 +489,7 @@ public void testStoreAndFetchWorkChunk_withOrWithoutGatedExecutionNoData_created
 		// execute & verify
 		String firstChunkId = storeFirstWorkChunk(JOB_DEFINITION_ID, FIRST_STEP_ID, instanceId, 0, null);
 		// mark the first chunk as COMPLETED to allow step advance
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImplTest.java:491]: Should this comment change?  Why is this backwards now?

File: hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/coordinator/ReductionStepExecutorServiceImpl.java [hunk 4]
```diff
@@ -340,10 +340,10 @@ private <PT extends IModelJson, IT extends IModelJson, OT extends IModelJson> vo
 		/*
 		 * Reduction steps are done inline and only on gated jobs.
 		 * As such, all workchunks once they get here should either be:
-		 * 1) READY (7.2 new status)
+		 * 1) REDUCTION_READY (7.2 new status)
 		 * 2) COMPLETED (7.0 legacy)
 		 */
-		if (theChunk.getStatus() != WorkChunkStatusEnum.READY
+		if (theChunk.getStatus() != WorkChunkStatusEnum.REDUCTION_READY
```
[the remark this answers is not in the document]
Comment (TipzCM) [reply hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/coordinator/ReductionStepExecutorServiceImpl.java:346]: fixed

File: hapi-fhir-storage-batch2/src/test/java/ca/uhn/fhir/batch2/coordinator/ReductionStepExecutorServiceImplTest.java (hunk 1 above)
Comment (TipzCM) [reply hapi-fhir-storage-batch2/src/test/java/ca/uhn/fhir/batch2/coordinator/ReductionStepExecutorServiceImplTest.java:80]: added

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImplTest.java (hunk 3 above)
Comment (TipzCM) [reply hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImplTest.java:491]: Changed the parameter order, as per gary's request.

It's still doing the same thing.

Now all the methods are `methodName(oldState, newStatE)`

Comment (TipzCM) [reply hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/batch2/JpaJobPersistenceImplTest.java:491]: Order has changed

**Pull request closed without merging**